From 3273d43ce26e3691db7fb203f046509d08f537e9 Mon Sep 17 00:00:00 2001
From: schspa <schspa@gmail.com>
Date: Fri, 26 Feb 2021 00:19:10 +0800
Subject: [PATCH 4/4] avb: Fix error when partition not found

part_get_info_by_name will return -1 on error, and >0 on success.

Signed-off-by: schspa <schspa@gmail.com>
Cc: Igor Opaniuk <igor.opaniuk@gmail.com>
Reviewed-by: Igor Opaniuk <igor.opaniuk@gmail.com>
---
 common/avb_verify.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/avb_verify.c b/common/avb_verify.c
index a2b73962..de386a9f 100644
--- a/common/avb_verify.c
+++ b/common/avb_verify.c
@@ -370,7 +370,7 @@ static struct mmc_part *get_partition(AvbOps *ops, const char *partition)
 	}
 
 	ret = part_get_info_by_name(mmc_blk, partition, &part->info);
-	if (!ret) {
+	if (ret < 0) {
 		printf("Can't find partition '%s'\n", partition);
 		goto err;
 	}
-- 
2.17.1

